src/components/DocsEntry.astro

```diff
@@ -0,0 +1,89 @@
+---
+import { getCollection } from "astro:content"
+import TableOfContents from "./TableOfContents.astro";
+import type { MarkdownHeading } from "astro";
+import DocLinkGroup from "./DocLinkGroup.astro";
+import type { CollectionEntry } from "astro:content";
+import DocsSwitchPageLink from "./DocsSwitchPageLink.astro";
+import TypographyContainer from "./TypographyContainer.astro";
+import NewsletterSignup from "./NewsletterSignup.astro";
+import Footer from "./Footer.astro";
+import { docsCollections, sectionAndSortEntries } from "../utils/docsCollections";
+import BarsIcon from "./icons/BarsIcon.astro";
+
+const docsEntries = await getCollection("docs", x => !x.id.includes("/runtime/"));
+
+const sidebarClass = "fixed overflow-auto w-[calc(var(--container-width)/5)] xl:w-[calc(var(--container-width)/6)] top-[3.5rem] h-[calc(100vh-3.5rem)] py-14";
+
+interface Props {
+  data: { title?: string | undefined };
+  section: string | null;
+  headings: MarkdownHeading[];
+  prev: CollectionEntry<"docs"> | undefined;
+  next: CollectionEntry<"docs"> | undefined;
+}
+
+const { data, section, headings, prev, next } = Astro.props;
+---
+
+<script>
+  let isSidebarOpen = false;
+  const sidebarLinks = document.getElementById("sidebar-docs-links")!;
+
+  document.getElementById("sidebar-open-docs-links")!.addEventListener("click", () => {
+    isSidebarOpen = !isSidebarOpen;
+    if (isSidebarOpen) {
+      sidebarLinks.classList.add("translate-x-0");
+      sidebarLinks.classList.remove("-translate-x-full");
+    } else {
+      sidebarLinks.classList.remove("translate-x-0");
+      sidebarLinks.classList.add("-translate-x-full");
+    }
+  });
+</script>
+
+<div class="relative z-30 sticky top-14 shadow-sm dark:shadow-purple-80 bg-white dark:bg-purple-90 lg:hidden">
+  <div class="container mx-auto flex items-center px-4 py-2">
+    <BarsIcon id="sidebar-open-docs-links" class="w-6 h-6 text-gray-40 dark:text-gray-60" />
+    <div class="text-sm truncate ml-3">{section ?? "Grain Docs"}&nbsp;&nbsp;<span class="text-color-dim-2">/</span>&nbsp;&nbsp;<span class="font-bold">{data.title}</span></div>
+  </div>
+</div>
+
+<div id="sidebar-docs-links" class="lg:hidden z-30 w-72 h-[calc(100vh-6rem)] fixed overflow-scroll top-24 left-0 p-3 bg-color-background shadow transition-transform -translate-x-full">
+  <h1 class="text-2xl font-semibold mb-4">Grain Docs</h1>
+
+  {docsCollections.map(coll => (
+    <DocLinkGroup title={coll.title} entries={sectionAndSortEntries(docsEntries.filter((x: any) => x.id.startsWith(coll.slugPrefix))).map(x => x.collectionEntry)} />
```

Comment:
Not sure what you mean?

Reply:
Basically if you look at the sidebar we currently have two `Random` libraries because one is `wasi/random` and one is `Random`, I think the best way we can really deal with this is by using the path as the name in the sidebar.


![Screenshot 2024-12-15 at 10 45 38 PM](https://github.com/user-attachments/assets/67fd2cc3-0105-4a5e-b059-48a53ec28db1)